### DISCUSSION TITLE
gwcgi: A rudimentary web-based interface

### DISCUSSION THREAD

**Discussion by ssterling** [keirf/greaseweazle#183] [Show and tell]
**gwcgi**, a web-based interface to Greaseweazle host tools.

Repositories: [GitLab](https://gitlab.com/ssterling/gwcgi), [GitHub](https://github.com/ssterling/gwcgi)

![](https://gitlab.com/ssterling/gwcgi/-/raw/master/screenshot.png)

**Rationale**: My Greaseweazle setup is a bit odd: I have the disk drives and the Greaseweazle itself all housed in an old computer separate from my main one (lacking in drive bays), and I access the whole thing via SSH.  However, I was no longer able to use the nifty templates feature from the [FMFF](https://github.com/FrankieTheFluff/FluxMyFluffyFloppy) GUI, which I had gotten quite used to.  Thus, I made this, a web-based approximation.

**Cool features**:
1. Two-click templates that fill in trackset parameters
2. File downloads automatically once finished
3. Can still be set up in ≈10 minutes, even in spite of the convoluted process

For the record: I'm by no means an experienced web designer nor am I even a competent JavaScript coder, but I’m just hoping that someone can benefit from my cobbled-together project.  I *do* plan on refactoring and adding features, though!

**Comment by keirf** [comment]
This is great. Just to let you know though that as of last night you can no longer run Greaseweazle straight out of the repository. It must now be installed as a proper python package. From the repository, 'make install' does something sensible. Download zips contain an sdist tarball you can pip install.

**Reply by ssterling** [reply]
Just updated the README and some of the files to reflect the change.
